packages/shared_preferences/android/src/main/java/io/flutter/plugins/sharedpreferences/SharedPreferencesPlugin.java

```diff
@@ -116,30 +117,47 @@ private String encodeList(List<String> list) throws IOException {
     return filteredPrefs;
   }
 
+  void backgroundTask(final Editor editor, final MethodChannel.Result result) {
+    new AsyncTask<Void, Void, Boolean>() {
+      @Override
+      protected Boolean doInBackground(Void... voids) {
+        return editor.commit();
+      }
+
+      @Override
+      protected void onPostExecute(Boolean value) {
+        super.onPostExecute(value);
+        result.success(value);
+      }
+    }.execute();
+  }
+
   @Override
   public void onMethodCall(MethodCall call, MethodChannel.Result result) {
     String key = call.argument("key");
-    boolean status = false;
+    Editor editor;
     try {
       switch (call.method) {
         case "setBool":
-          status = preferences.edit().putBoolean(key, (boolean) call.argument("value")).commit();
+          editor = preferences.edit().putBoolean(key, (boolean) call.argument("value"));
+          backgroundTask(editor, result);
           break;
         case "setDouble":
           double doubleValue = ((Number) call.argument("value")).doubleValue();
           String doubleValueStr = Double.toString(doubleValue);
-          status = preferences.edit().putString(key, DOUBLE_PREFIX + doubleValueStr).commit();
+          editor = preferences.edit().putString(key, DOUBLE_PREFIX + doubleValueStr);
+          backgroundTask(editor, result);
```

Comment:
Once you change method name, you can do:

`commitAsync(preferences.edit().putString(...), result);`